Mathlib/Data/Matroid/Dual.lean

```diff
@@ -0,0 +1,264 @@
+    and_iff_left ((inter_subset_left _ _).trans (diff_subset _ _))]
+  refine' fun B' hB' ↦ by_contra (fun hem ↦ _)
+  rw [nonempty_iff_ne_empty, not_ne_iff, ←union_singleton, diff_inter_diff,
+   inter_distrib_right, union_empty_iff, singleton_inter_eq_empty, diff_eq,
+   inter_right_comm, inter_eq_self_of_subset_right hB'.subset_ground, ←diff_eq,
+   diff_eq_empty] at hem
+  obtain ⟨f, hfb, hBf⟩ := hB.exchange hB' ⟨he.2, hem.2⟩
+
+  have hi : M.Indep (insert f (B ∩ X))
+  · refine' hBf.indep.subset (insert_subset_insert _)
+    simp_rw [subset_diff, and_iff_right (inter_subset_left _ _), disjoint_singleton_right,
+      mem_inter_iff, iff_false_intro he.1.2, and_false, not_false_iff]
+  exact hfb.2 (hBX.mem_of_insert_indep (Or.elim (hem.1 hfb.1) (False.elim ∘ hfb.2) id) hi).1
+
+theorem Base.inter_basis_iff_compl_inter_basis_dual (hB : M.Base B) (hX : X ⊆ M.E := by aesop_mat):
+    M.Basis (B ∩ X) X ↔ M﹡.Basis ((M.E \ B) ∩ (M.E \ X)) (M.E \ X) := by
+  refine' ⟨hB.compl_inter_basis_of_inter_basis, fun h ↦ _⟩
+  simpa [inter_eq_self_of_subset_right hX, inter_eq_self_of_subset_right hB.subset_ground] using
+    hB.compl_base_dual.compl_inter_basis_of_inter_basis h
+
+theorem base_iff_dual_base_compl (hB : B ⊆ M.E := by aesop_mat) :
+    M.Base B ↔ M﹡.Base (M.E \ B) := by
+  rw [dual_base_iff, diff_diff_cancel_left hB]
+
+theorem ground_not_base (M : Matroid α) [h : RkPos M﹡] : ¬M.Base M.E := by
+  rwa [rkPos_iff_empty_not_base, dual_base_iff, diff_empty] at h
+
+theorem Base.ssubset_ground [h : RkPos M﹡] (hB : M.Base B) : B ⊂ M.E :=
+  hB.subset_ground.ssubset_of_ne (by rintro rfl; exact M.ground_not_base hB)
+
+theorem Indep.ssubset_ground [h : RkPos M﹡] (hI : M.Indep I) : I ⊂ M.E := by
+  obtain ⟨B, hB⟩ := hI.exists_base_superset; exact hB.2.trans_ssubset hB.1.ssubset_ground
+
+/-- A coindependent set of `M` is an independent set of the dual of `M﹡`. we give it a separate
+  definition to enable dot notation. Which spelling is better depends on context. -/
+abbrev Coindep (M : Matroid α) (I : Set α) : Prop := M﹡.Indep I
+
+theorem coindep_def : M.Coindep X ↔ M﹡.Indep X := Iff.rfl
+
+theorem Coindep.indep (hX : M.Coindep X) : M﹡.Indep X :=
+  hX
+
+@[simp] theorem dual_coindep_iff : M﹡.Coindep X ↔ M.Indep X := by
+  rw [Coindep, dual_dual]
+
+theorem Indep.coindep (hI : M.Indep I) : M﹡.Coindep I :=
+    dual_coindep_iff.2 hI
+
+theorem coindep_iff_exists' : M.Coindep X ↔ (∃ B, M.Base B ∧ B ⊆ M.E \ X) ∧ X ⊆ M.E := by
+  simp_rw [Coindep, dual_indep_iff_exists', and_comm (a := _ ⊆ _), and_congr_left_iff, subset_diff]
+  exact fun _ ↦ ⟨fun ⟨B, hB, hXB⟩ ↦ ⟨B, hB, hB.subset_ground, hXB.symm⟩,
+    fun ⟨B, hB, _, hBX⟩ ↦ ⟨B, hB, hBX.symm⟩⟩
+
+theorem coindep_iff_exists (hX : X ⊆ M.E := by aesop_mat) :
+    M.Coindep X ↔ ∃ B, M.Base B ∧ B ⊆ M.E \ X := by
+  rw [coindep_iff_exists', and_iff_left hX]
+
+theorem coindep_iff_subset_compl_base : M.Coindep X ↔ ∃ B, M.Base B ∧ X ⊆ M.E \ B := by
```

Comment:
`\` is usually called `sdiff`:
```suggestion
theorem coindep_iff_subset_sdiff_base : M.Coindep X ↔ ∃ B, M.Base B ∧ X ⊆ M.E \ B := by
```

Reply:
If you make sure that `M.E \ x` is documented to be `compl`, then that seems fine to me. I'd almost go as far as define the abbreviation `M.compl B` for this, but that might make things more difficult to work with.